synapse_domain_rule_checker/__init__.py

```diff
@@ -73,7 +73,16 @@ async def _is_new_room(self, room_id: str) -> bool:
 
         events = await self._api.get_room_state(room_id, state_event_filter)
 
-        room_creator = events[(EventTypes.Create, "")].sender
+        create_event = events.get((EventTypes.Create, ""))
+        if create_event is None:
+            # If we don't have a create event for this room then it means we're processing
+            # an invite we've received over federation, for a room we're not yet in. So
+            # we can't pull the data we need to check whether the room is new or not.
+            # In this case, we assume the remote homeserver, which knows the state of the
+            # room, has done the necessary checks and has vetted this invite.
```

Comment:
Remote joins via a given server already place a great deal of trust in that server, so this sounds sane to me.